File: test17_out/pollen.time/TimerMilli/TimerMilli.c

/*
 * ======== string literals ========
 */

static const char pollen_time_TimerMilli__s__fileName[] = "TimerMilli.p";
static const char pollen_time_TimerMilli__s__0[] = "TimerMilli";

/*
 * ======== function definitions ========
 */

bool pollen_time_TimerMilli_running__E() {
    return(  (pollen_time_TimerMilli_duration__V) != 0 && pollen_interfaces_TimerDriver_running__E()  );
}

void pollen_time_TimerMilli_stop__E() {
    pollen_interfaces_TimerDriver_stop__E();
    pollen_time_TimerMilli_duration__V = 0;
}

void pollen_time_TimerMilli_start__E( pollen_time_TimerMilli_Handler_handle h, uint16 duration, bool repeat ) {
    pollen_time_TimerMilli_clientTick__V = (h);
    pollen_time_TimerMilli_count__V = 0;
    pollen_time_TimerMilli_duration__V = (duration);
    pollen_time_TimerMilli_repeat__V = (repeat);
    pollen_interfaces_TimerDriver_start__E();
}

void pollen_time_TimerMilli_targetInit__I() {
    pollen_interfaces_TimerDriver_setHandler__E((pollen_time_TimerMilli_timerTick__E));
}

void pollen_time_TimerMilli_timerTick__E() {
    ++(pollen_time_TimerMilli_count__V);
    if ((pollen_time_TimerMilli_count__V) == (pollen_time_TimerMilli_duration__V)) {
        if ((pollen_time_TimerMilli_clientTick__V) != null) {
            pollen_time_TimerMilli_clientTick__V();
        }
        (pollen_time_TimerMilli_count__V) = 0;
        if (!(pollen_time_TimerMilli_repeat__V)) {
            pollen_time_TimerMilli_stop__E();
        }
    }
}

File: test17_out/pollen.time/TimerMilli/TimerMilli.js

/*
 * ======== MODULE pollen.time.TimerMilli ========
 */

var $$c = function() {
    const TimerMilli = {};
    TimerMilli.TimerMilli = TimerMilli
    TimerMilli.$name = 'pollen.time.TimerMilli';
    TimerMilli.pollen$used = 0;
    TimerMilli.targetInit = new $$Ref('pollen_time_TimerMilli_targetInit__I');
    TimerMilli.start = new $$Ref('pollen_time_TimerMilli_start__E');
    TimerMilli.stop = new $$Ref('pollen_time_TimerMilli_stop__E');
    TimerMilli.running = new $$Ref('pollen_time_TimerMilli_running__E');
    TimerMilli.timerTick = new $$Ref('pollen_time_TimerMilli_timerTick__E');
    TimerMilli.bindTimerDriver = function( td ) {
        var $$text = '';
        TimerMilli.timer = (td);
        return $$text;
    }
    TimerMilli.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    TimerMilli.pollen__uses__ = function() {
    }
    TimerMilli.$$privateInit = function() {
        TimerMilli.timer = 0;
        TimerMilli.count = 0;
        TimerMilli.duration = 0;
        TimerMilli.repeat = true;
        TimerMilli.clientTick = 0;
    }
    return TimerMilli;
}
var $$u = $$c();
$units['pollen.time.TimerMilli'] = $$u;
$units.push($$u);

